components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,773 @@
+            }
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST,
+                    "mandated parameter (JWTID) does not available.");
+        }
+
+        if (StringUtils.isBlank(claimsSet.getJWTID())) {
+            // (jti) is a blank value.
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                        claimsSet.getIssuer() + ".The request has invalid values for the parameter (jti).");
+            }
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST,
+                    "invalid value for the parameter (JWTID)");
+        }
+    }
+
+    /**
+     * Checks whether the request is properly signed.
+     *
+     * @param signedJWT SignedJWT.
+     * @return Boolean.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     */
+    private Boolean isValidSignature(SignedJWT signedJWT) throws CibaAuthFailureException {
+
+        String alg = signedJWT.getHeader().getAlgorithm().getName();
+        if (StringUtils.isEmpty(alg)) {
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, "Algorithm must not be null.");
+        }
+
+        if (alg.startsWith("RS")) {
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, "Algorithms not supported.");
+        }
+        return true;
+    }
+
+    /**
+     * Checks whether the audience is valid as expected.
+     *
+     * @param claimsSet CIBA Authentication request as claim sets.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     */
+    private void validateAudience(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
+
+        List<String> aud = claimsSet.getAudience();
+        String clientId = claimsSet.getIssuer();
+
+        try {
+            // Validation for aud-audience.
+            if (aud.isEmpty()) {
+                // No value for audience found in the request.
+                if (log.isDebugEnabled()) {
+                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                            claimsSet.getIssuer() + ".The request is missing the mandatory parameter 'aud'.");
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, " missing (aud) parameter.");
+            }
+
+            OAuthAppDO appDO = OAuth2Util.getAppInformationByClientId(clientId);
+            String domain = OAuth2Util.getTenantDomainOfOauthApp(appDO);
+
+            // Getting expected audience dynamically from configs.
+            String expectedAudience = OAuth2Util.getIdTokenIssuer(domain);
+
+            // Validation for aud-audience to meet mandated value.
+            if (!aud.contains(expectedAudience)) {
+                // The audience value does not suit mandated value.
+                if (log.isDebugEnabled()) {
```

Comment:
We should mention the expected audience value here so that it will easy to debug.